iota/api.py

```diff
@@ -22,948 +23,948 @@
+
+    def __init__(self, adapter, testnet=False):
+        # type: (AdapterSpec, bool) -> None
+        """
+        :param adapter:
+          URI string or BaseAdapter instance.
+
+        :param testnet:
+          Whether to use testnet settings for this instance.
+        """
+        super(StrictIota, self).__init__()
+
+        if not isinstance(adapter, BaseAdapter):
+            adapter = resolve_adapter(adapter)
+
+        self.adapter = adapter  # type: BaseAdapter
+        self.testnet = testnet
+
+    def __getattr__(self, command):
+        # type: (Text) -> BaseCommand
+        """
+        Creates a pre-configured command instance.
+
+        This method will only return commands supported by the API class.
+
+        If you want to execute an arbitrary API command, use
+        :py:meth:`create_command`.
+
+        :param command:
+          The name of the command to create.
+
+        References:
+          - https://iota.readme.io/docs/making-requests
+        """
+        # Fix an error when invoking :py:func:`help`.
+        # https://github.com/iotaledger/iota.lib.py/issues/41
+        if command == '__name__':
+            # noinspection PyTypeChecker
+            return None
+
+        try:
+            command_class = self.commands[command]
+        except KeyError:
+            raise InvalidCommand(
+                '{cls} does not support {command!r} command.'.format(
+                    cls=type(self).__name__, command=command,),
+            )
+
+        return command_class(self.adapter)
+
+    def create_command(self, command):
+        # type: (Text) -> CustomCommand
+        """
+        Creates a pre-configured CustomCommand instance.
+
+        This method is useful for invoking undocumented or experimental
+        methods, or if you just want to troll your node for awhile.
+
+        :param command:
+          The name of the command to create.
+        """
+        return CustomCommand(self.adapter, command)
+
+    @property
+    def default_min_weight_magnitude(self):
+        # type: () -> int
+        """
+        Returns the default ``min_weight_magnitude`` value to use for API
+        requests.
+        """
+        return 9 if self.testnet else 14
+
+    def add_neighbors(self, uris):
+        # type: (Iterable[Text]) -> dict
+        """
+        Add one or more neighbors to the node.  Lasts until the node is
+        restarted.
+
+        :param uris:
+          Use format ``udp://<ip address>:<port>``.
+          Example: ``add_neighbors(['udp://example.com:14265'])``
+
+        References:
+          - https://iota.readme.io/docs/addneighors
+        """
+        return core.AddNeighborsCommand(self.adapter)(uris=uris)
+
+    def attach_to_tangle(
+        self,
+        trunk_transaction,
+        branch_transaction,
+        trytes,
+        min_weight_magnitude=None
+    ):
+        # type: (TransactionHash, TransactionHash, Iterable[TryteString], int) -> dict  # noqa
```

Comment:
❔ Does this guy need a `# noqa`?

Reply:
According to flake8, yes. The line has more than 80 character. I was not sure if these lines are somewhat interpreted by sphinx so I didn't break that line into two shorter ones. Would that be an option instead?

Reply:
Gotcha.  Those are type hints, used by some IDEs (e.g., PyCharm) for autocompletion, etc.

It is possible to address the line length issue here, by doing this:

```python
def attach_to_tangle(
    self,
    trunk_transaction,  # type: TransactionHash
    branch_transaction,  # type: TransactionHash
    trytes,  # type: Iterable[TryteString]
    min_weight_magnitude=None  # type: Optional[int]
):
+        # type: (...) -> dict
```

See [PEP-484](https://www.python.org/dev/peps/pep-0484/#suggested-syntax-for-python-2-7-and-straddling-code) for more info.

Reply:
That looks neat! :+1: